tests/integration/rosetta/rosetta_test.go

```diff
@@ -0,0 +1,253 @@
+// +build integration
+
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package rosetta_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/labstack/echo/v4"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/identifier"
+	"github.com/optakt/flow-dps/service/state"
+
+	"github.com/optakt/flow-dps/rosetta/invoker"
+	"github.com/optakt/flow-dps/rosetta/retriever"
+	"github.com/optakt/flow-dps/rosetta/validator"
+)
+
+// initialize the rosetta Data HTTP handler once and reuse for all tests.
+var rosettaSvc *rosetta.Data
+
+func TestMain(m *testing.M) {
+
+	const testDbBackupFile = "rosetta-get-balance-test.db"
+	opts := badger.DefaultOptions("").
+		WithInMemory(true).
+		WithReadOnly(true).
+		WithLogger(nil)
+
+	db, err := badger.Open(opts)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	dbSnapshot, err := os.Open(testDbBackupFile)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	defer dbSnapshot.Close()
+
+	err = db.Load(dbSnapshot, 10)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	core, err := state.NewCoreFromDB(db)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	// setup scaffolding for minimal server we need for the tests
+	chain := flow.ChainID("flow-testnet").Chain()
+	validate := validator.New(chain, core.Height())
+	headers := invoker.NewHeaders(core.Chain())
+	invoke := invoker.New(zerolog.Nop(), core, chain, headers)
+	retrieve := retriever.New(invoke)
+
+	rosettaSvc = rosetta.NewData(validate, retrieve)
+
+	os.Exit(m.Run())
+}
+
+func TestGetBalance(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		wantStatusCode int
+		request        rosetta.BalanceRequest
+		wantBalance    string
+		wantHandlerErr assert.ErrorAssertionFunc
```

Comment:
I'd recommend grouping inputs and outputs separately for the sake of readability

```suggestion
		request        rosetta.BalanceRequest
		
		wantStatusCode int
		wantBalance    string
		wantHandlerErr assert.ErrorAssertionFunc
```